### PR TITLE
Tests to cover member & client-side ICacheManagers.

### PR DESCRIPTION
**Pull request by vbekiaris**
Code coverage improvements for `HazelcastInstanceCacheManager` and `ClientICacheManager`

### CODE REVIEW

File: hazelcast-client/src/test/java/com/hazelcast/client/impl/ClientICacheManagerTest.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+import static org.mockito.Mockito.when;
+
+/**
+ * Unit test ClientICacheManager.getCache exception handling
+ */
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelTest.class})
+public class ClientICacheManagerTest {
+
+    @Test(expected = HazelcastException.class)
```
Comment (tombujok) [inline hazelcast-client/src/test/java/com/hazelcast/client/impl/ClientICacheManagerTest.java]: ExpectedException JUnit rule is much better in this context IMHO, but it's part of personal coding style.
Right now you don't know which line throws the exception. With the ExpectedException rule you also don't have to use the `fail` statement.

Comment (vbekiaris) [inline hazelcast-client/src/test/java/com/hazelcast/client/impl/ClientICacheManagerTest.java]: 👍 I adopted the `ExpectedException` rule style for both tests. Using `fail` was redundant in the first place, but in any case I agree `ExpectedException` rule style makes for clearer declaration of intent in the code.

**Comment by tombujok** [comment]
👍

**Comment by Donnerbart** [comment]
:+1:

**Pull request merged**